packages/core/src/components/discord-message/discord-message.tsx

```diff
@@ -96,48 +104,56 @@ export class DiscordMessage implements ComponentInterface {
 		const profileData: Profile = Reflect.get(profiles, this.profile) ?? {};
 		const profile: Profile = { ...defaultData, ...profileData, ...{ avatar: resolveAvatar(profileData.avatar ?? this.avatar) } };
 
-		// @ts-expect-error ts doesn't understand this
-		const highlightMention: boolean = Array.from(this.el.children).some((child: HTMLDiscordMentionElement): boolean => {
-			return child.tagName.toLowerCase() === 'discord-mention' && child.highlight && child.type !== 'channel';
-		});
+		const highlightMention: boolean =
+			// @ts-expect-error ts doesn't understand this
+			Array.from(this.el.children).some((child: HTMLDiscordMentionElement): boolean => {
+				return child.tagName.toLowerCase() === 'discord-mention' && child.highlight && ['user', 'role'].includes(child.type);
```

Comment:
Isn't the array here missing channel now?

Reply:
I'm pretty sure this only selects mentions that would make sense to highlight, like user mentions and role mentions. The previous code would reject highlighted channel mentions.
```ts
const highlightMention: boolean = Array.from(this.el.children).some((child: HTMLDiscordMentionElement): boolean => {
			return child.tagName.toLowerCase() === 'discord-mention' && child.highlight && child.type !== 'channel';
		});
```

Reply:
Ah right I see now. Used to be accurate, but of course since you added `voice` and `locked` it no longer was.